static/js/modules/tables.js

```diff
@@ -604,15 +609,38 @@ DataTable.prototype.fetchSuccess = function(resp) {
 
 DataTable.prototype.fetchError = function() {
   var self = this;
+  var errorMessage = '<div class="filter__message filter__message--error">' +
+      '<strong>We had trouble processing your request</strong><br>' +
+      'Please try again. If you still have trouble, ' +
+      '<button class="js-filter-feedback">let us know</button></div>';
 
-  setTimeout(function() {
-    $('.is-loading').removeClass('is-loading').addClass('is-unsuccessful');
-    self.$processing.hide();
-  }, helpers.LOADING_DELAY);
+  $('.filter__message').remove();
+
+  // text search error message
+  if ($(updateChangedEl).attr('type') === 'text' && $(updateChangedEl).hasClass('tt-input') === false) {
+    $(updateChangedEl).parent().after($(errorMessage));
+  }
+  else {
+    $('label.is-loading')
+      .removeClass('is-loading')
+      .addClass('is-unsuccessful')
+      .after($(errorMessage))
+      .hide().fadeIn();
+  }
+
+  $('button.is-loading').removeClass('is-loading');
+
+  self.$processing.hide();
 
-  setTimeout(function() {
+  clearTimeout(messageTimer);
+
+  messageTimer = setTimeout(function() {
```

Comment:
Should the error messages stay for each input, or only to the last input that triggered an error?